### PR TITLE
[release-1.14]: Configure the securityContext, resources and pod annotations in helm

### PR DESCRIPTION
**Pull request by houshengbo**
Fixed: #1869

### CODE REVIEW

File: config/charts/knative-operator/templates/operator.yaml [hunk 1]
```diff
@@ -48,6 +48,10 @@ metadata:
     app.kubernetes.io/component: operator-webhook
     app.kubernetes.io/version: "{{ .Chart.Version }}"
     app.kubernetes.io/name: knative-operator
+  {{- if and .Values.knative_operator.operator_webhook.annotations }}
```
Comment (yuzisun) [inline config/charts/knative-operator/templates/operator.yaml:51]: I think the annotations under the metadata is not necessarily the same as the annotations on the template/metadata.

Comment (houshengbo) [reply config/charts/knative-operator/templates/operator.yaml]: Just remove this section.

File: config/charts/knative-operator/templates/operator.yaml [hunk 2]
```diff
@@ -114,14 +130,23 @@ spec:
           value: knative.dev/operator
         - name: KUBERNETES_MIN_VERSION
           value: "{{ .Values.knative_operator.kubernetes_min_version }}"
+        {{- if and .Values.knative_operator.operator_webhook.securityContext }}
+        securityContext:
```
Comment (yuzisun) [inline config/charts/knative-operator/templates/operator.yaml:130]: There is also another securityContext on the pod level

Comment (houshengbo) [reply config/charts/knative-operator/templates/operator.yaml:130]: but the yaml does not have it be default, if .Values.knative_operator.operator_webhook.securityContext is not available, do not add it.

**Review by yuzisun — APPROVED** [review]

**Pull request merged**